test/googlesource_test.dart

```diff
@@ -0,0 +1,164 @@
+library googlesource_test;
+
+import 'dart:async';
+import 'dart:convert';
+
+import '../tool/src/googlesource.dart';
+import '../tool/src/simple_http_client.dart';
+import 'package:mock/mock.dart';
+import 'package:test/test.dart';
+
+void main() => defineTests();
+
+void defineTests() {
+  group('GoogleSourceFile', () {
+    GoogleSourceFile file;
+
+    void testHtmlConversion(List<String> lines) {
+      file = new GoogleSourceFile(asHtml(lines), 'example.com');
+
+      expect(file.fileContents, lines.join('\n'));
+    }
+
+    test('correctly parses simple raw html', () {
+      testHtmlConversion(['just one line']);
+    });
+
+    test('correctly parses multiline raw html', () {
+      testHtmlConversion(['multiple', 'small', 'lines']);
+    });
+
+    test('correctly parses multiline raw html with whitespace', () {
+      testHtmlConversion(['if (true)', '  goto label;', '', '', 'label']);
+    });
+
+    test('unescapes files', () {
+      var testEscapeString = 'this & that is <\"\'>';
+      var escapedHtmlFile = '<table><tr><td></td><td><a name="1"></a><span>'
+          '${HTML_ESCAPE.convert(testEscapeString)}</span></td></tr></table>';
+      file = new GoogleSourceFile(escapedHtmlFile, 'www.example.com');
+
+      expect(file.fileContents, testEscapeString);
+    });
+  });
+
+  group('GoogleSourceCrawler', () {
+    var baseUri = 'http://www.example.com/';
+    MockSimpleHttpClient client;
+    GoogleSourceCrawler crawler;
+
+    setUp(() {
+      client = new MockSimpleHttpClient();
+      crawler = new GoogleSourceCrawler(baseUri, client: client);
+
+      client.when(callsTo('getHtmlAtUri'))
+        ..thenReturn(new Future.value())
+        ..alwaysReturn(new Future.value(''));
+    });
+
+    test('returns correct files in single directory', () async {
+      prepopulateHttpResponses(client, test1);
+
+      var files = await crawler.findAllMatchingFiles('test').toList();
+
+      expect(files.length, 3);
+      expect(files[0].url, '/test/a.idl');
+      expect(files[1].url, '/test/b.idl');
+      expect(files[2].url, '/test/c.idl');
+    });
+
+    test('correctly follows the file tree', () async {
+      prepopulateHttpResponses(client, test2);
+
+      var files = await crawler.findAllMatchingFiles('test').toList();
+
+      expect(files.length, 1);
```

Comment:
I believe iterable.single throws if there is more than one so you dont need to verify (apply this everywhere below)